Tools/LyTestTools/ly_test_tools/cli/codeowners_hint.py

```diff
@@ -0,0 +1,232 @@
+        current_path = target_path
+
+    for _ in range(1000):
+        codeowners_path = os.path.join(current_path, ".github", "CODEOWNERS")
+        if os.path.exists(codeowners_path):
+            return codeowners_path
+        next_path = os.path.dirname(current_path)
+        if next_path == current_path:
+            break  # reached filesystem root
+        current_path = next_path
+
+    logger.warning(f"No GitHub CODEOWNERS file found in a GitHub repo which contains {target_path}")
+    return ""
+
+
+def get_codeowners_from(target_path: str, codeowners_path: str) -> (str, str):
+    """
+    Fetch ownership information matching the target filesystem path from a CODEOWNERS file
+    :param target_path: path to match in the GitHub CODEOWNERS file
+    :param codeowners_path: path to CODEOWNERS file
+    :return: tuple of (matched_path_entry, matched_owner_aliases), which will be empty when nothing was matched.
+        The aliases will also be empty when a matched path is explicitly unowned.
+    """
+    if not os.path.isfile(codeowners_path):
+        logger.warning(f"No GitHub CODEOWNERS file found at {codeowners_path}")
+        return "", ""
+
+    if os.path.getsize(codeowners_path) > _GITHUB_CODEOWNERS_BYTE_LIMIT:
+        logger.warning(f"GitHub CODEOWNERS file found at {codeowners_path} exceeds the standard limit of "
+                       f"{_GITHUB_CODEOWNERS_BYTE_LIMIT} bytes")
+        return "", ""
+
+    # operate only on unix-style separators
+    repo_root = pathlib.PurePosixPath(codeowners_path.replace('\\', '/')).parent.parent
+    unix_normalized_target = pathlib.PurePosixPath(target_path.replace('\\', '/'))
+    try:  # is_relative_to introduced in Python 3.9
+        repo_relative_target = unix_normalized_target.relative_to(repo_root)
+        repo_rooted_target = pathlib.PurePosixPath("/" + str(repo_relative_target))  # relative_to removes leading slash
+    except ValueError:
+        logger.warning(f"Path '{target_path}' is not inside the repo of GitHub CODEOWNERS file {codeowners_path}")
+        return "", ""
+
+    with open(codeowners_path) as codeowners_file:
+        # GitHub syntax only applies the final matching rule ==> parse in reverse order and take first match
+        for line in reversed(list(codeowners_file)):
+            clean_line = line.strip()
+            if clean_line and not clean_line.startswith('#'):  # ignore blanks and full-line comments
+                # entry format should be "owned/path/     @alias1 @alias2 user@example.com @aliasN..."
+                split_entry = line.split(maxsplit=1)
+                owned_path = split_entry[0]
+                if _codeowners_path_matches(repo_rooted_target, owned_path):
+                    if len(split_entry) > 1:
+                        aliases = split_entry[1].split("#", maxsplit=1)[0].strip()  # remove trailing comment
+                    else:  # explicitly unowned entry with no comment
+                        aliases = ""
```

Comment:
Maybe we should explicitly say "Unowned" here?

Reply:
It will later default to the default owner param